### PR TITLE
Remove Host Provisioning references

### PR DESCRIPTION
**Pull request by gmcculloug**
Remove any `MiqHostProvisionRequest` and `HostProvision` records.

Followup from https://github.com/ManageIQ/manageiq/pull/17604

### CODE REVIEW

**Comment by gmcculloug** [comment]
@miq-bot add_label technical debt
@miq-bot assign @Fryguy

File: db/migrate/20180712122000_remove_host_provisioning.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  end
+
+  class MiqRequestTask < ActiveRecord::Base
+    self.inheritance_column = :_type_disabled
+  end
+
+  def up
+    say_with_time("Delete Host Provision Requests and Tasks") do
+      MiqRequest.where(:type => :MiqHostProvisionRequest).destroy_all
+      MiqRequestTask.where(:type => :MiqHostProvision).destroy_all
```
Comment (Fryguy) [inline db/migrate/20180712122000_remove_host_provisioning.rb]: Are there any relationships that need to be cleaned up as well?

Comment (gmcculloug) [reply db/migrate/20180712122000_remove_host_provisioning.rb]: After reviewing the `MiqRequest` model I added `miq_approvals`.  I believe that is everything.

**Comment by gmcculloug** [comment]
Updated to use `class_name` on the associations.

**Pull request merged**